packages/babel-core/src/transformation/file/index.js

```diff
@@ -410,8 +412,32 @@ export default class File extends Store {
   }
 
   parse(code: string) {
+    let parseCode = parse;
+    let parserOpts = this.opts.parserOpts;
+
+    if (parserOpts.parser) {
+      if (typeof parserOpts.parser === "string") {
+        let dirname = parserOpts.dirname || process.cwd();
```

Comment:
Ah you're right can access this.opts.filename: maybe `let dirname = parserOpts.dirname || path.dirname(this.opts.filename);`